libraries/chain/apply_context.cpp

```diff
@@ -131,12 +141,15 @@ void apply_context::exec_one()
       } else {
          act_digest = digest_type::hash(*act);
       }
-   } catch( const fc::exception& e ) {
-      action_trace& trace = trx_context.get_action_trace( action_ordinal );
-      trace.error_code = controller::convert_exception_to_error_code( e );
-      trace.except = e;
-      finalize_trace( trace, start );
+   } catch ( const std::bad_alloc& ) {
+      throw;
+   } catch ( const boost::interprocess::bad_alloc& ) {
       throw;
+   } catch( const fc::exception& e ) {
+      handle_exception(e);
+   } catch ( const std::exception& e ) {
```

Comment:
Why was catch of std::exception added here? Do we really want all std::exceptions to be ignored? Either this is not needed OR this is a consensus change since nodes without this code will fail transaction differently.

Reply:
Would it be better to catch just the specific std errors being thrown by variant and optional?  I can change the `std::exception` handler to `std::bad_variant_access` and whatever specific std exceptions that are thrown.

Reply:
I think what you have is likely fine. I would only argue that someone does a thorough review of possible exceptions in code paths where this was added. I suspect there are not any other std exceptions we really need to worry about. So just need someone to review and give approval.